core/store/src/main/java/uk/gov/gchq/gaffer/store/schema/Schema.java

```diff
@@ -123,6 +128,49 @@ public boolean isAggregationEnabled() {
         return isEnabled;
     }
 
+    /**
+     * Creates a Function that takes and element as input and outputs a set of objects including the
+     * Group-by values, the Identifiers and the group. These act as a key and can be used in a
+     * Collector. This function is dependent on the hashcode of the key components so make sure the
+     * Identifiers and Group-By objects all have a valid hashcode method.
+     *
+     * @return a Set which makes up a unique Identifier. The set is composed of the input Element's
+     * Group-By, Identifiers and Group.
+     */
+    public Function<Element, Set<Object>> createGroupByFunction() {
```

Comment:
Oh, I've just been thinking about this a bit more and we can't just use a Set as the key as we lose the context as to what things are in the key. For example:
if Entity 1 = {
group: "Red"
vertex: "1"
properties: {}
}

Entity2 = {
group: "1"
vertex: "Red"
properties: {}
}

These entities are different but when Red and 1 are put in the Set they would have the same key.

Instead of using a Set perhaps we should just use the Element class.

For example, something like:

```java
public Function<Element, Element> toElementKey(final Schema schema) {
        return element -> {
            final Element key = element.emptyClone();
            final SchemaElementDefinition elDef = schema.getElement(key.getGroup());
            if (elDef == null) {
                throw new RuntimeException("received element " + element + " which belongs to a " +
                        "group not found in the schema");
            }

            final Set<String> groupBy = elDef.getGroupBy();
            for (final String property : groupBy) {
                key.putProperty(property, element.getProperty(property));
            }
            return key;
        };
    }
```